src/Microsoft.OpenApi.OData.Reader/Common/Utils.cs

```diff
@@ -222,5 +223,66 @@ internal static void AddCustomAttributesToExtensions(this IDictionary<string, IO
 
             return atrributesValueMap;
         }
+
+        /// <summary>
+        /// Checks whether the base type of an <see cref="IEdmStructuredType"/> is referenced as a type within the Edm model.
+        /// </summary>
+        /// <param name="model">The Edm model.</param>
+        /// <param name="baseType">The base type of the target <see cref="IEdmStructuredType"/>.</param>
+        /// <param name="structuredTypes">Optional: The IEnumerable of <see cref="IEdmStructuredType"/> to check against.</param>
+        /// <param name="complexTypes">Optional: The IEnumerable of <see cref="IEdmComplexType"/> to check against.</param>
+        /// <param name="actions">Optional: The IEnumerable of <see cref="IEdmAction"/> to check against.</param>
+        /// <returns>True if reference is found, otherwise False.</returns>
+        internal static bool IsBaseTypeReferencedAsTypeInModel(
+            this IEdmModel model,
+            IEdmStructuredType baseType,
+            IEnumerable<IEdmStructuredType> structuredTypes = null,
+            IEnumerable<IEdmComplexType> complexTypes = null,
+            IEnumerable<IEdmAction> actions = null)
+        {
+            string baseTypeName = baseType?.FullTypeName();
+            bool isBaseTypeEntity = Constants.EntityName.Equals(baseTypeName?.Split('.').Last(), StringComparison.OrdinalIgnoreCase);
+
+            if (!string.IsNullOrEmpty(baseTypeName) && !isBaseTypeEntity)
+            {
+                structuredTypes ??= model.GetAllElements()
+                        .Where(static x => x.SchemaElementKind == EdmSchemaElementKind.TypeDefinition)
+                        .Where(static y => !y.Name.Equals(Constants.EntityName, StringComparison.OrdinalIgnoreCase))
+                        .OfType<IEdmStructuredType>();
+
+                complexTypes ??= model.GetAllElements()
+                    .Where(static x => x.SchemaElementKind == EdmSchemaElementKind.TypeDefinition)
+                    .OfType<IEdmComplexType>();          
```

Comment:
```suggestion       
```

One last thing, I think we don't need this now as `IEdmStructuredType` is implemented by `IEdmComplexType`. So, the `structuredTypes` collection should already include complex types. 


https://github.com/OData/odata.net/blob/master/src/Microsoft.OData.Edm/Schema/Interfaces/IEdmComplexType.cs

Reply:
Quite true. Thanks